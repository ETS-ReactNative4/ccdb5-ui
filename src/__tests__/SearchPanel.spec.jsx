import ReduxSearchPanel, { SearchPanel } from '../SearchPanel';
import configureMockStore from 'redux-mock-store'
import React from 'react'
import renderer from 'react-test-renderer'
import thunk from 'redux-thunk'
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router';
import { shallow } from 'enzyme';

function setupEnzyme(initialProps={}) {
  const props = Object.assign({
    lastUpdated: new Date(2016 , 1, 1)
  }, initialProps)

  const target = shallow(<SearchPanel {...props} />);

  return {
    target,
    props
  }
}

describe('SearchPanel', () => {
    let target, props
    beforeEach(() => {
      ({ target, props } = setupEnzyme())
    })

    describe('renders', () => {
      it('renders with last updated date', () => {
        expect(target).toMatchSnapshot()
      })
    })
  })