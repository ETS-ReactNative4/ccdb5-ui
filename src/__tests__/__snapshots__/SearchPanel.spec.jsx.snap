exports[`SearchPanel renders renders with last updated date 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="search-panel">
    <h2>
        Search complaint data 
        <span
            className="date-subscript">
            (last updated: 
            <FormattedDate
                value={2016-02-01T05:00:00.000Z} />
            )
        </span>
    </h2>
    <Connect(SearchBar) />
    <Connect(Component) />
</div>,
  "nodes": Array [
    <div
      className="search-panel">
      <h2>
            Search complaint data 
            <span
                  className="date-subscript">
                  (last updated: 
                  <FormattedDate
                        value={2016-02-01T05:00:00.000Z} />
                  )
            </span>
      </h2>
      <Connect(SearchBar) />
      <Connect(Component) />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SearchPanel
        lastUpdated={2016-02-01T05:00:00.000Z} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SearchPanel {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "lastUpdated": 2016-02-01T05:00:00.000Z,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="search-panel">
          <h2>
                    Search complaint data 
                    <span
                              className="date-subscript">
                              (last updated: 
                              <FormattedDate
                                        value={2016-02-01T05:00:00.000Z} />
                              )
                    </span>
          </h2>
          <Connect(SearchBar) />
          <Connect(Component) />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="search-panel">
          <h2>
                    Search complaint data 
                    <span
                              className="date-subscript">
                              (last updated: 
                              <FormattedDate
                                        value={2016-02-01T05:00:00.000Z} />
                              )
                    </span>
          </h2>
          <Connect(SearchBar) />
          <Connect(Component) />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SearchPanel
    lastUpdated={2016-02-01T05:00:00.000Z} />,
}
`;
